paf-mvp-core-js/src/model/io-signature.service.ts

```diff
@@ -0,0 +1,63 @@
+import { IDSASigner } from '@core/crypto/digital-signature';
+import { GetIdsPrefsResponse, IdsPrefsResponse, PostIdsPrefsResponse } from './generated-model';
+import { Unsigned } from './model';
+
+const SIGN_SEP = '\u2063';
+
+/**
+ * Service for signing requests and response of endpoints
+ */
+export interface IIO_DSASignService {
+  /**
+   * Sign Ids-and-preferences response.
+   */
+  signIdAndPrefsResponse(response: Unsigned<IdsPrefsResponse>): Promise<IdsPrefsResponse>;
```

Comment:
One thing I'm concerned about is how you will **split** the areas related to different messages.

For example, only the operator will ever sign `IdsPrefsResponse`, while only the operator clients will ever sign `IdsPrefsResquest`.

Do you plan to put all messages in this single interface? I worry that this will be confusing to have the ability, for example, to create a SignService and sign an `IdsPrefsResquest` from the operator, while in fact this will never be a real-life use-case.

Do you consider creating one service for the operator and one for the client, for instance?

The question applies to `IO_DSAValidator` as well, of course.

Reply:
If I understand your concern well, I think the existing design is already like that: when you import a module from the core, you get every class from the operator and the client. And this is why I struggled a lot doing this change.

Of course, we can do a separation. And if you go in this way, I suggest that the respective classes go in their separate apps (i.e client-node, operator-node).

Good to include in the technical vision but to postpone with an iterative approach.

Reply:
> when you import a module from the core, you get every class from the operator and the client

but you import one class at a time, so you are able to cherry pick.

My concern here is more about a single class having a bunch of methods.

> I suggest that the respective classes go in their separate apps (i.e client-node, operator-node).

yes!